include/bluetooth/audio/cap.h

```diff
@@ -0,0 +1,352 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_AUDIO_CAP_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_AUDIO_CAP_H_
+
+/**
+ * @brief Common Audio Profile (CAP)
+ *
+ * @defgroup bt_cap Common Audio Profile (CAP)
+ *
+ * @ingroup bluetooth
+ * @{
+ *
+ * [Experimental] Users should note that the APIs can change
+ * as a part of ongoing development.
+ */
+
+#include <zephyr/types.h>
+#include <bluetooth/audio/csis.h>
+#include <bluetooth/audio/tbs.h>
+#include <bluetooth/audio/media_proxy.h>
+#include <bluetooth/audio/audio.h.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Type of CCID records */
+enum bt_cap_ccid_record_type {
+	/** CCID belongs to a Telephone Bearer Service instance */
+	BT_CAP_CCID_TYPE_TELEPHONE_BEARER,
+	/** CCID belongs to a Media Control Service instance */
+	BT_CAP_CCID_TYPE_MEDIA_PLAYER
+};
+
+/** CCID record structure */
+struct bt_cap_ccid_record {
+	/** The type of the record */
+	enum bt_cap_ccid_record_type type;
+
+	union {
+		/** Reference to a local or remote Telephone Bearer instance */
+		struct bt_tbs *telephone_bearer;
+		/** Reference to a local or remote Media Control instance */
+		struct media_player *media_player;
+	};
+};
+
+/** Callback structure for CAP procedures */
+struct bt_cap_cb {
+	/**
+	 * @brief CCID find callback
+	 *
+	 * Callback for bt_cap_find_ccid_service().
+	 * This is called for the first service instance found for the
+	 * specified CCID.
+	 *
+	 * @param conn        Connection object used in
+	 *                    bt_cap_find_ccid_service().
+	 * @param err         Error value. 0 on success, GATT error on positive
+	 *                    value or errno on negative value.
```

Comment:
Inconsistent language.
Proposal: "Error value.  0 on success.  GATT error (positive values) or errno (negative value) on failure."
Or (shorter): "Error value.  0 on success.  GATT error or errno on failure."

(Also other places)

Reply:
One thing we could do (and perhaps for the better) is to avoid ever returning GATT error codes in API functions, which would fix all the above issues. 

Will update.

Reply:
Borrowed text from `bluetooth.h` and `gatt.h`. Please re-review